### PR TITLE
HDF5DataSet subset (formerly index)

### PR DESCRIPTION
**Pull request by DaniBodor**
renamed and fixed issues regarding this parameter

### CODE REVIEW

File: docs/quickstart.md [hunk 1]
```diff
@@ -53,7 +53,7 @@ metrics_exporters = [OutputExporter(metrics_output_directory),
 dataset = HDF5DataSet(
     root = "./",
     database = database,
-    index = range(400),
+    subset = range(400),
```
Comment (gcroci2) [inline docs/quickstart.md:56]: We should insert keys here, but let's just remove `subset` from the quickstart file for now.

**Review by gcroci2 — COMMENTED** [review]
Great! Another parameter's usage clarified.
I would suggest to also add a unit test in tests/test_dataset.py file since we are never testing the `subset` parameter when it's not None. You could define a test in which you open the hdf5 file, save the keys in a variable, select only n keys and pass them to `subset`. Then assert if the number of the keys you selected equals the dataset length.

**Comment by DaniBodor** [comment]
@gcroci2 , does it automatically update my pull request with the new stuff or do I need to do something?

**Comment by gcroci2** [comment]
> @gcroci2 , does it automatically update my pull request with the new stuff or do I need to do something?

It does it automatically when you commit to the branch of the PR :)

**Comment by gcroci2** [comment]
You can also see your last commit above "added unit test for subset parameter" @DaniBodor

**Comment by DaniBodor** [comment]
OK!
Was I supposed to have created a separate branch and PR from this one, so that you can quickly check what has changed?
I guess because it was only few small changes it's fine, but for bigger changes would that make sense?

**Comment by gcroci2** [comment]
> OK! Was I supposed to have created a separate branch and PR from this one, so that you can quickly check what has changed? I guess because it was only few small changes it's fine, but for bigger changes would that make sense?

In general, if the changes are related to the same issue/PR it's fine to commit them in the same branch/PR, like you did.

**Pull request merged**